nixos/modules/services/web-apps/tuliprox.nix

```
@@ -0,0 +1,252 @@
+                      username = "local";
+                      device = "hdhr1";
+                    }
+                  ];
+                  filter = "!final_channel_lineup!";
+                  options = {
+                    ignore_logo = false;
+                    share_live_streams = true;
+                  };
+                  mapping = [
+                    "iptv-org"
+                  ];
+                }
+              ];
+            }
+          ];
+        }
+      '';
+      description = ''
+        Source definitions
+
+        Refer to the [Tuliprox documentation] for available attributes
+
+        [Tuliprox documentation]: https://github.com/euzu/tuliprox?tab=readme-ov-file#2-sourceyml
+      '';
+    };
+
+    apiProxySettings = lib.mkOption {
+      type = settingsFormat.type;
+      example = ''
+        {
+          server = [
+            {
+              name = "default";
+              protocol = "http";
+              host = "192.169.1.9";
+              port = 8901;
+              timezone = "Europe/Paris";
+              message = "Welcome to tuliprox";
+            }
+            {
+              name = "external";
+              protocol = "https";
+              host = "tuliprox.mydomain.tv";
+              port = 443;
+              timezone = "Europe/Paris";
+              message = "Welcome to tuliprox";
+            }
+          ];
+          user = [
+            {
+              target = "xc_m3u";
+              credentials = [
+                {
+                  username = "test1";
+                  password = "secret1";
+                  token = "token1";
+                  proxy = "reverse";
+                  server = "default";
+                  exp_date = 1672705545;
+                  max_connections = 1;
+                  status = "Active";
+                }
+              ];
+            }
+          ];
+        }
+      '';
+      description = ''
+        Users and proxy configuration
+
+        Refer to the [Tuliprox documentation] for available attributes
+
+        [Tuliprox documentation]: https://github.com/euzu/tuliprox?tab=readme-ov-file#3-api-proxy-config
+      '';
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    services.tuliprox.systemSettings = {
+      api = {
+        host = lib.mkDefault "127.0.0.1";
+        port = lib.mkDefault 8901;
+        web_root = lib.mkDefault "${cfg.package}/web";
+      };
+      working_dir = lib.mkDefault "\${env:STATE_DIRECTORY}/data";
+      backup_dir = lib.mkDefault "\${env:STATE_DIRECTORY}/backup";
+      custom_stream_response_path = lib.mkDefault "${cfg.package}/resources";
+    };
+    services.tuliprox.sourceSettings.sources = lib.mkDefault [ ];
+    services.tuliprox.apiProxySettings = {
+      server = lib.mkDefault [
+        {
+          name = "default";
+          protocol = "http";
+          host = cfg.systemSettings.api.host;
+          port = cfg.systemSettings.api.port;
+          timezone = "Europe/Paris";
```

This seems like a weird default. How about

```suggestion
          timezone = config.time.timeZone or "Etc/UTC";
```